Src/AutoFixture.xUnit.net.UnitTest/CompositeDataAttributeInsufficientDataTest.cs

```diff
@@ -74,17 +74,7 @@ public IEnumerator<object[]> GetEnumerator()
             yield return CreateTestCase(
                 data: new[]
                 {
-                    new FakeDataAttribute(this.method, this.parameterTypes, new[] {  new object[] { 1, 2    },  new object[] {  3,  4     }, new object[] { 5, 6 } }),
-                    new FakeDataAttribute(this.method, this.parameterTypes, new[] {  new object[] { 7, 8, 9 },  new object[] { 10, 11, 12 }                        })
-                }
-            );
-
-            yield return CreateTestCase(
-                data: new[]
-                {
-                    new FakeDataAttribute(this.method, this.parameterTypes, new[] { new object[] {  1,  2     }, new object[] {  3,  4     }, new object[] { 5, 6 } }),
-                    new FakeDataAttribute(this.method, this.parameterTypes, new[] { new object[] {  7,  8,  9 }, new object[] { 10, 11, 12 }                        }),
-                    new FakeDataAttribute(this.method, this.parameterTypes, new[] { new object[] { 13, 14, 15 }                                                     })
+                    new FakeDataAttribute(this.method, this.parameterTypes, new[] { new object[] { 1, 2, 3, 4 } }),

```

Comment:
It's a tricky one. You could ask the same of xUnit - why does it throw an exception, rather than just discard the excess? I think the answer is correctness: if a `DataAttribute` is providing data then it *knows* the type signature of the method and the number of parameters to provide. If you're receiving too many objects then there must be a bug in one of your attributes. If there's a bug, it's better to alert the user than silently ignore it.

Perhaps a compromise is to just return what you are given - don't trim the results, but don't throw an exception either. Let xUnit worry about what to do with an oversized data set.